src/main/java/com/google/cloud/teleport/templates/common/JdbcConverters.java

```diff
@@ -107,7 +116,31 @@ public TableRow mapRow(ResultSet resultSet) throws Exception {
       TableRow outputTableRow = new TableRow();
 
       for (int i = 1; i <= metaData.getColumnCount(); i++) {
-        outputTableRow.set(metaData.getColumnName(i), resultSet.getObject(i));
+        if (resultSet.getObject(i) == null) {
+          outputTableRow.set(metaData.getColumnName(i), resultSet.getObject(i));
+          continue;
+        }
+
+        /*
+         * DATE:      EPOCH MILLISECONDS -> yyyy-MM-dd
+         * DATETIME:  EPOCH MILLISECONDS -> yyyy-MM-dd hh:mm:ss.SSSSSS
+         * TIMESTAMP: EPOCH MILLISECONDS -> yyyy-MM-dd hh:mm:ss.SSSSSSXXX
+         *
+         * MySQL drivers have ColumnTypeName in all caps and postgres in small case
+         */
+        switch (metaData.getColumnTypeName(i).toLowerCase()) {
+          case "date":
+            outputTableRow.set(metaData.getColumnName(i), df.format(resultSet.getObject(i)));
+            break;
+          case "datetime":
+            outputTableRow.set(metaData.getColumnName(i), dtf.format(Timestamp.valueOf((LocalDateTime)resultSet.getObject(i))));
```

Comment:
Is there any specific reason why converting to Timestamp.valueOf which also requires LocalDateTime conversion.

Reply:
done